Add quantile feature to forestplot ridgeplot

percygautam opened the pull request:
## Description
<!--
Thank you so much for your PR!  To help us review your contribution, please
consider the following points:

- The PR title should summarize the changes, for example "Add new group argument for the
  pair plot".  Avoid non-descriptive titles such as "Addresses issue #348". If your pull 
  request addresses an issue, please use the pull request title to describe 
  the issue and mention the issue number in the pull request description.

- The description should provide at least 1-2 sentences describing the pull request
  in detail (Why is this change required?  What problem does it solve?) and
  link to any relevant issues. If modifying a plot, render your plot to inspect for changes 
  and copy image in the pull request message on Github

- Please prefix the title of incomplete contributions with [WIP] (to indicate a work in 
  progress). WIPs may be useful to (1) indicate you are working on something to avoid 
  duplicated work, (2) request broad review of functionality or API, or (3) seek collaborators.
-->

Fixes #756 . Add new feature quantiles to ridgeplot of forestplot.
## Checklist
<!-- Feel free to remove check-list items aren't relevant to your change -->

- [x] Does the PR follow [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) 
      PR format?
- [x] Has included a sample plot to visually illustrate the changes? (only for plot-related functions)
- [x] Is the new feature properly documented with an example?
- [x] Is the code style correct (follows pylint and black guidelines)?
- [x] Is the change listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased )?

<!--
Also, please consider reading the contributing guidelines and code of conduct carefully before submitting the PR. They are available at 
- https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md
- https://github.com/arviz-devs/arviz/blob/master/CODE_OF_CONDUCT.md

We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or
the recommended next step seems overly demanding, if you would like help in
addressing a reviewer's comments, or if you have been waiting too long to hear
back on your PR.
-->

percygautam commented:
Currently implemented for `matplotlib`. It looks something like:
![Screenshot 2020-02-06 at 1 06 16 AM](https://user-images.githubusercontent.com/20489158/73878293-4fcbee00-4880-11ea-8d60-62254983bb54.png)

There is a problem of having step size too small in plots. `ax.fill_between` will inverse fill between `x[i]` and `x[i+1]`, this cause the quantile line to be very thin. (almost disappear in more dense step sized plots like `defs Italy`). How can we clearly show these quantiles?

aloctavodia commented:
As a workaround you can stretch the figure along the x-axis, something like `figsize=(14, 4)`. Another solution is to re-think how we display quantiles for kdes. Maybe instead of not filling the area we could use two alternating shades (using two different alpha values) ¿?

percygautam commented:
@aloctavodia I tinkered a little with figsize generation, and below size seems okayish:
```
    if figsize is None:
        if ridgeplot_quantiles is None:
            figsize = (min(12, sum(width_ratios) * 2), plot_handler.fig_height())
        else:
            figsize = (min(12, sum(width_ratios) * 5), plot_handler.fig_height() * 1.2)
```
Graph looks like this :
![Screenshot 2020-02-11 at 2 34 41 AM](https://user-images.githubusercontent.com/20489158/74190004-88503b00-4c77-11ea-8787-a266e855e10c.png)

About alternate shades for quantiles, Should I add that also in this? Should alternate shades also be added to normal kdeplot with quantiles?

aloctavodia commented:
Not sure, but I guess a good solution is to work on this first and after merge work on a separated PR for the shade feature. For the next PR we may even think on a general function that is called by the kdeplot and the ridgeplot.

aloctavodia commented:
I do not think the function should change the figsize automatically, the user should be in control of the figsize

percygautam commented:
Okay, Let's get this one done first. I have changed the figsize for ridgeplot_quantiles, if it is not explicitly defined by the user. It's creating a figure of `(12, 5.2)`. Is this correct or should we specify some different default figsize for quantiles plot?

percygautam commented:
I have added  the quantile feature to bokeh `ridgeplot`, and changed the default figsize to clearly show the quantiles (for [0.2, 0.8]):
![Screenshot 2020-02-12 at 4 42 36 PM](https://user-images.githubusercontent.com/20489158/74330712-2c410000-4db8-11ea-810e-bf0334c286e9.png)

Also, The figsize of normal ridgeplot (without quantiles) is not normalized, making the bokeh figure to look cluttered. Should I change that also?

aloctavodia commented on arviz/plots/backends/bokeh/forestplot.py:
```diff
@@ -58,7 +59,10 @@ def plot_forest(
     )
 
     if figsize is None:
-        figsize = (min(12, sum(width_ratios) * 2), plot_handler.fig_height())
```
ohh, sorry I misread your previous example. This is OK

ahartikainen commented on arviz/plots/backends/matplotlib/forestplot.py:
```diff
@@ -484,7 +505,7 @@ def treeplot(self, qlist, credible_interval):
     def ridgeplot(self, mult, ridgeplot_kind):
         """Get data for each ridgeplot for the variable."""
-        xvals, yvals, pdfs, colors = [], [], [], []
+        xvals, yvals, pdfs, pdfs_q, colors = [], [], [], [], []
```
this could be 

    xvals = yvals = pdfs = pdfs_q = colors = []

but it is not that important, so no need to change it
